src/status.ts

```diff
@@ -0,0 +1,267 @@
+      const status: vscode.LanguageStatusItem = this.createStatusItem(
+        "addMissingGem",
+        "Ruby LSP: Bundle Add",
+        vscode.LanguageStatusSeverity?.Error
+      );
+
+      this.context.subscriptions.push(
+        vscode.commands.registerCommand("ruby-lsp.addMissingGem", () => {
+          status.text = "Ruby LSP: Adding gem...";
+          status.command = undefined;
+          status.busy = true;
+          addGem()
+            .then(() => status.dispose())
+            .catch(() => {});
+        })
+      );
+
+      status.command = {
+        title: "Run bundle add and install",
+        command: "ruby-lsp.addMissingGem",
+      };
+      return true;
+    }
+    return false;
+  }
+
+  private async activateGemOutdatedButton() {
+    const gemOutdated = await isGemOutdated();
+    if (!gemOutdated) {
+      return;
+    }
+
+    const commandId = "updateOutdatedGem";
+
+    const status: vscode.LanguageStatusItem = this.createStatusItem(
+      "updateGem",
+      "Ruby LSP: The gem is not up-to-date",
+      vscode.LanguageStatusSeverity?.Warning
+    );
+
+    this.context.subscriptions.push(
+      vscode.commands.registerCommand(commandId, async () => {
+        status.text = "Ruby LSP: Updating Ruby LSP";
+        status.busy = true;
+        status.command = undefined;
+
+        const result = await updateGem();
+
+        if (result.stderr.length > 0) {
+          status.text = "Ruby LSP: Update failed";
+          status.command = {
+            title: "Try again",
+            command: commandId,
+          };
+
+          if (
+            result.stderr.includes(
+              "Bundler attempted to update ruby-lsp but its version stayed the same"
+            )
+          ) {
+            vscode.window.showWarningMessage(
+              "Could not update the ruby-lsp gem. Is the version in the Gemfile pinned?"
+            );
+          } else {
+            vscode.window.showErrorMessage("Failed to update gem.");
+          }
+        } else {
+          status.dispose();
+        }
+      })
+    );
+
+    status.command = {
+      title: "Update",
+      command: commandId,
+    };
+  }
+
+  private createYjitStatus(ruby: Ruby) {
+    const useYjit = vscode.workspace.getConfiguration("rubyLsp").get("yjit");
+
+    let [major, minor]: number[] = [0, 0];
+
+    if (this.ruby.rubyVersion) {
+      [major, minor] = this.ruby.rubyVersion.split(".").map(Number);
+    }
+
+    const yjit: vscode.LanguageStatusItem =
+      vscode.languages.createLanguageStatusItem("yjit", this.selector);
+    yjit.name = "Ruby LSP Status";
+
+    if (useYjit && ruby.yjitEnabled && [major, minor] >= [3, 2]) {
+      yjit.text = "Ruby LSP: YJIT is in use";
+    } else {
+      yjit.text = "Ruby LSP: YJIT is not in use";
+      if ([major, minor] >= [3, 2]) {
+        yjit.severity = vscode.LanguageStatusSeverity?.Warning;
+        yjit.command = {
+          title: "Enable it",
+          command: "",
+        };
```

Comment:
That would depend on why YJIT is disabled. If it's just the configuration, then it's easy to offer to enabled that. However, if their Ruby was compiled without YJIT because they don't have the Rust compiler in their machine, then it might be trickier and, honestly, territory we don't want to venture into.

How about displaying enable it only if the configuration is set to false when YJIT is available and Ruby is > 3.2?